velero-csi-plugin/csi_snapshotter_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"testing"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	corev1api "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestSetPVCAnnotationsAndLabels(t *testing.T) {
+	testCases := []struct {
+		name         string
+		pvc          corev1api.PersistentVolumeClaim
+		snapshotName string
+		backupName   string
+	}{
+		{
+			name: "should set annotations and labels when they are nil",
+			pvc: corev1api.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: nil,
+					Labels:      nil,
+				},
+			},
+		},
+		{
+			name: "should set annotation and labels on existing annotatation and labels",
+			pvc: corev1api.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"foo": "foo-val",
+						"bar": "bar-val",
+						"baz": "baz-val",
+					},
+					Labels: map[string]string{
+						"foo": "foo-val",
+						"bar": "bar-val",
+						"baz": "baz-val",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		before := tc.pvc.DeepCopy()
+		setPVCAnnotationsAndLabels(&tc.pvc, tc.snapshotName, tc.backupName)
+
+		if tc.pvc.Annotations == nil {
```

Comment:
For conditions like this, we can use `"github.com/stretchr/testify/assert"`. It has helpers like `assert.NotNil` and `assert.True`. IMO, that dependency is worthwhile.

Reply:
Yes. I have another PR in the pipeline to switch to `stretchr` for all tests.

Reply:
Ah, ok, then I'll remove the request for changes and we can get them all in one.